src/lazy/expanded/sequence.rs

```diff
@@ -333,3 +399,52 @@ fn expand_next_sequence_value<'top, D: Decoder>(
         }
     }
 }
+
+fn expand_next_sequence_value_from_resolved<'top, D: Decoder>(
+    environment: Environment<'top, D>,
+    evaluator: &mut MacroEvaluator<'top, D>,
+    iter: &mut impl Iterator<Item = IonResult<ValueExpr<'top, D>>>,
+) -> Option<IonResult<LazyExpandedValue<'top, D>>> {
```

Comment:
🗺️ This generic function (and the one above) will also work for `make_list` when that gets implemented.